### DISCUSSION TITLE
Quarkus redirects creates new trace and span and I can't link them as a parent

### DISCUSSION THREAD

**Discussion by ukrcoder** [open-telemetry/opentelemetry-java-instrumentation#9940] [Q&A]
Using quarkus 3.4.3 and latest opentelemetry there. I want whole my flow be linked and to see it inside zipkin/jager (no matter). But whan I have redirects in my code -> than I
ll always have separate track and new traceId flow.
I cannot always pass traceparent header because of ui (in that case secured cookies could be used so I can store state). But adding as a header doesn't help to link to the parent. 

Tried to add parent to the context with next code
`String[] values = traceparentCookie.split("-");

            SpanContext parentContext = SpanContext.create(values[1], values[2], TraceFlags.getSampled(), TraceState.getDefault());
            Span parentSpan = Span.wrap(parentContext);
            Log.infof("TRACEPARENT DATA: trace is = " + values[1] + " and span is = " + values[2]);

            Span span = tracer.spanBuilder("ui-provider-picker").setParent(Context.current().with(parentSpan)).startSpan();
            try (Scope scope = span.makeCurrent()) {
            } finally {
                span.setAttribute("test", "12345");
                span.end();
            }`

it creates new span and attaches to base process as a last one and then starts new trace with needed me endpoint. It's also created as an empty one without tags or any info.

Overall what I want: to link one span with another one retrieving all needed tags. Couldn't find solution meanwhile.
Any suggestions?

**Comment by jkwatson** [comment]
I'm going to transfer this to the instrumentation repository, since it seems to be related to how to write quarkus-friendly instrumentation for redirects.
